packages/maskbook/src/plugins/Ethereum/UI/GasNowDialog.tsx

```diff
@@ -0,0 +1,232 @@
+    const usdRate = nativeToken?.price?.usd
+    const options = useMemo(
+        () => [
+            {
+                title: t('plugin_gas_now_dialog_standard'),
+                type: 'standard',
+                gasPrice: gasNow?.standard ?? 0,
+                time: t('plugin_gas_now_dialog_min', { time: 3 }),
+            },
+            {
+                title: t('plugin_gas_now_dialog_fast'),
+                type: 'fast',
+                gasPrice: gasNow?.fast ?? 0,
+                time: t('plugin_gas_now_dialog_min', { time: 1 }),
+            },
+            {
+                title: t('plugin_gas_now_dialog_custom'),
+                type: 'custom',
+                gasPrice: customGasToWei,
+                time: t('plugin_gas_now_dialog_sec', { time: 15 }),
+            },
+        ],
+        [gasNow, customGas],
+    )
+
+    const validationMessage = useMemo(() => {
+        if (select === 2 && customGasToWei === 0) return t('plugin_gas_now_dialog_zero_gas')
+        return ''
+    }, [customGasToWei, options, select])
+
+    const onConfrim = useCallback(() => {
+        currentGasPriceSettings.value = select == 2 ? customGasToWei : options[select].gasPrice
+        const type = options[select].type as keyof GasNow
+        setDialog({ open: false, type })
+    }, [customGasToWei, options, select, setDialog])
+
+    return (
+        <InjectedDialog
+            open={open}
+            onClose={closeDialog}
+            title={t('plugin_gas_now_dialog_title')}
+            DialogProps={{ maxWidth: 'xl' }}>
+            <DialogContent className={classes.content}>
+                <List className={classes.list}>
+                    {options.map((option, i) => (
+                        <ListItem
+                            onClick={() => (i === 2 || gasNow) && setSelect(i)}
+                            className={classNames(
+                                classes.box,
+                                i === select ? classes.boxSelected : undefined,
+                                i !== 2 && !gasNow ? classes.disableBox : undefined,
+                            )}
+                            key={i}>
+                            <div className={classes.head}>
+                                <Typography className={classes.title}>{option.title}</Typography>
+                                <FiberManualRecordIcon
+                                    className={classNames(classes.dot, i === select ? classes.dotSelector : undefined)}
+                                />
+                            </div>
+                            {gasNow || i === 2 ? (
+                                <>
+                                    <div>
+                                        {i == 2 ? (
+                                            <div className={classes.gweiBox}>
+                                                <TextField
+                                                    className={classes.customInput}
```

Comment:
Do we need to set a max value for this field?

Reply:
I don't think it is needed